crates/core/tedge/src/cli/connect/command.rs

```diff
@@ -219,13 +212,88 @@ impl ConnectCommand {
                 .auth_method
                 .is_basic(&c8y_config.credentials_path);
             if !use_basic_auth && !self.offline_mode && connection_check_success {
-                let _ = self.tenant_matches_configured_url(config).await;
+                let _ = self.tenant_matches_configured_url().await;
             }
             enable_software_management(&bridge_config, &*self.service_manager).await;
         }
 
         Ok(())
     }
+
+    /// Validate that the new certificate is actually accepted by the cloud endpoint
+    ///
+    /// Return:
+    /// - Ok(true) when the new certificate has been promoted as the current one
+    /// - Ok(false) when it's safer to keep the current certificate untouched
+    /// - Err(err) when the endpoint is not correctly configured
+    async fn validate_new_certificate(
+        &self,
+        bridge_config: &BridgeConfig,
+        certificate_shift: CertificateShift,
+    ) -> Result<bool, ConfigError> {
+        if self.offline_mode {
+            println!("Offline mode. Skipping new device certificate validation");
+            println!(
+                "  => use current certificate {}",
+                &certificate_shift.active_cert_path
+            );
+            println!(
+                "  => ignoring new certificate {}",
+                &certificate_shift.new_cert_path
+            );
+            return Ok(false);
+        }
+
+        let spinner = Spinner::start(format!(
+            "Checking new certificate validity: {}",
+            &certificate_shift.new_cert_path
+        ));
+
+        let res = if bridge_config.cloud_name.eq("c8y") {
+            let use_basic_auth = false;
+            let device_type = &self.config.device.ty;
+            let profile_name = self.cloud.profile_name().map(|p| p.as_ref());
+            let mut mqtt_auth_config = self.config.mqtt_auth_config_cloud_broker(profile_name)?;
+            if let Some(client_config) = mqtt_auth_config.client.as_mut() {
+                client_config.cert_file = certificate_shift.new_cert_path.to_owned()
+            }
+
+            create_device_with_direct_connection(
+                use_basic_auth,
```

Comment:
This `use_basic_auth` seems always false. You can use `bridge_config.auth_type` to check if it is Basic or not.

Reply:
`use_basic_auth` is set `false` on purpose here, because if we reach this point this is because there are two certificates: a current one + a new one to be validated.